issue 10, no print functionality

zpallin opened the pull request:
At the request of a code for san jose member I added "no print" functionality here.

zpallin committed:
issue #10 : fixing gitignore to include *.swp

zpallin committed:
issue #10 : new static button for "noprint" and modified css

zpallin committed:
issue #10 : added not_printing functionality and loading page

In this way it acts like the printing page except no printing is sent...

Kyle-Falconer commented:
I think this is fine, but this pull request looks like it only includes the HTML elements. I think the Python also needs a new route to only submit, but not print. 

See: https://github.com/codeforsanjose/CFSJ-Login-System/blob/master/nametag_server.py#L51

zpallin commented:
@Kyle-Falconer I see what you mean and I was considering doing that originally but decided not to for some reason. I'll make the change.

zpallin committed:
issue #10 : change name of route from print and funciton from upload_file to "submit"

These changed are made to reflect the need for the functionality of the route/function to match its naming

zpallin committed:
issue #10 : yet another revisioning; the name should be even more understanding to its purpose. signing > submit

zpallin commented:
Okay, see my latest change.

Kyle-Falconer commented:
I don't want to block this change, but I think it could be improved by still having a single "thank you" type page.

Maybe, can we pass in another form value to indicate whether printing should happen or not, and then load the thank you page with a query parameter with one of three values, which displays the appropriate message via javascript:

* printed successfully
* chose not to print
* printing failed because of error

If you don't feel like doing this, let me know and I'll merge in your pull request, and I can modify it at some other time.

zpallin commented:
Except it does have a "thank you" type page already.

zpallin committed:
issue #10 : instead render template with passed config

In order to reduce duplicates, I turned the thankyou page into a
template and pass a message into it on render based on
circumstances.

zpallin committed:
issue #10 : clean up comment

zpallin commented:
Instead of displaying the message via javascript, I am rendering the html in a template form with the correct data. Flask handles this pretty well. I removed the excess routes and static files.

We won't need to handle for an error condition since there are only two ways to submit data -- testing this code would reveal that an "else" condition would not be utilized unless we added a button to the form that did not have a name value, so we don't need that.

Otherwise, I think I've satisfied your recommendations. If you think it's good, I also have the power to merge code apparently. :P
